### ISSUE TITLE
[BUG] boolean value not mapped correctly when using querybuilder

### ISSUE THREAD

**Issue by Seemspyo** [typeorm/typeorm#7103]
## Issue Description
Boolean type parameters are not correctly mapped when using querybuilder.

### example
```ts
// get blocked users
const query = getRepository(User)
.createQueryBuilder('user')
.where('user.blocked = :blocked', { blocked: true });

console.log(query.getQueryAndParameters().slice(1)); // get parameters
```

### output
```text
[ true ]
```

### Expected Behavior

IMO the output should be:
```text
[ 1 ]
```

### Relevant Database Driver(s)

- [ ] `aurora-data-api`
- [ ] `aurora-data-api-pg`
- [ ] `better-sqlite3`
- [ ] `cockroachdb`
- [ ] `cordova`
- [ ] `expo`
- [ ] `mongodb`
- [ x ] `mysql`
- [ ] `nativescript`
- [ ] `oracle`
- [ ] `postgres`
- [ ] `react-native`
- [ ] `sap`
- [ ] `sqlite`
- [ ] `sqlite-abstract`
- [ ] `sqljs`
- [ ] `sqlserver`

### Are you willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time, and I know how to start.
- [ ] Yes, I have the time, but I don't know how to start. I would need guidance.
- [ x ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by nebkat** [comment]
Does it cause issues elsewhere? `Driver.preparePersistentValue()` is used by the UPDATE/INSERT query builders for values, but isn't used for WHERE conditions, so maybe that should be changed.

**Comment by Seemspyo** [comment]
@nebkat
Apparently this doens't causing issues.
But people who don't know typeorm uses `tinyint(1)` for boolean might confuse about this. And I think this is what orm should taking care of.

**Comment by nebkat** [comment]
It's not just TypeORM, MySQL itself doesn't support `BOOLEAN`, it maps it to `tinyint(1)` and I believe `true`/`false` will always get converted to `1`/`0` anyway?

There might be other situations where we should be using `Driver.preparePersistentValue()` in `WHERE`, e.g. `Date`s, so this might come about later anyway, but if you have no problems here I guess you can can close the issue for now.

**Comment by Seemspyo** [comment]
I'm sorry, but I didn't get your point.

I know mysql does not support Boolean. So I can easily figure out TypeORM is the culprit. I can just use `1` instead `true` and pretend nothing was wrong.

But if you define `@Column` as boolean, I believe you think "Oh, I can also use bool values in where clause?".

As you say, this may not a BIG issue for the experts. But apparently this is a bug for those who are not.

**Comment by nebkat** [comment]
I'm sorry I also don't get your point then :grin:

> Apparently this doens't causing issues.

I was asking if it is a problem that TypeORM sends `true` to the driver. Is there a crash, does the query not work, i.e. is it actually a bug? Or are you just concerned that it's sending `true` instead of `1`, but nothing bad is happening?

You should of course be able to use bool values in the where clause since this makes sense on the ORM side of things (and in general the user should not care about what driver is in the background).

**Comment by Seemspyo** [comment]
Yep. The query is not working when set `{ blocked: true }`, but works fine with `{ blocked: 1 }`.

If I rather paste a `sql` it is more clear but I find out `getSql()` not resolve parameters. Sorry if you confused.
